qiskit_braket_provider/providers/adapter.py

```diff
@@ -477,8 +452,83 @@ def convert_qiskit_to_braket_circuits(
     Returns:
         Circuit (Iterable[Circuit]): Braket circuit
     """
-    for circuit in circuits:
-        yield convert_qiskit_to_braket_circuit(circuit)
+    warnings.warn(
+        "convert_qiskit_to_braket_circuits() is deprecated and "
+        "will be removed in a future release. "
+        "Use to_braket() instead. ",
+        DeprecationWarning,
+    )
+    return to_braket(circuits)
+
+
+@singledispatch
+def from_braket(circuit: Any) -> Any:
+    """Return a Qiskit quantum circuit from a Braket quantum circuit.
+     Args:
+            circuit (Circuit | List[Circuit]): Braket Quantum Cricuit
+
+    Returns:
+        QuantumCircuit | List[QuantumCircuit]: Qiskit quantum circuit
+    """
+    raise TypeError(f"Cannot convert {circuit.__class__} to Qiskit circuit.")
+
+
+@from_braket.register
+def _(circuit: Circuit) -> QuantumCircuit:
+    """Return a Qiskit quantum circuit from a Braket quantum circuit.
+     Args:
+            circuit (Circuit): Braket Quantum Cricuit
+
+    Returns:
+        QuantumCircuit: Qiskit quantum circuit
+    """
+
+    qiskit_circuit = QuantumCircuit(max(circuit.qubits) + 1)
+    dict_param = {}
+    for instruction in circuit.instructions:
+        gate_name = instruction.operator.name.lower()
+        gate_instance = gate_name_to_qiskit_gate.get(gate_name, None)
```

Comment:
This work feels like one abstraction layer lower -- so I think there's a good opportunity here to factor out a private function for grabbing `gate_cls, gate_params`.